h5py/tests/test_dataset.py

```diff
@@ -468,10 +468,12 @@ def test_compound(self):
         self.assertAlmostEqual(dset[4], v)
 
     def test_exc(self):
-        """ Bogus fill value raises ValueError """
-        with self.assertRaises(ValueError):
-            dset = self.f.create_dataset('foo', (10,),
-                    dtype=[('a', 'i'), ('b', 'f')], fillvalue=42)
+        """ Bogus fill value converts correctly"""
+        dt = np.dtype([('a', 'i'), ('b', 'f')])
+        v = np.array([42], dtype=dt)[0]
+        dset = self.f.create_dataset('foo', (10,), dtype=[('a', 'i'), ('b', 'f')], fillvalue=42)
```

Comment:
To my mind, this probably *should* be an error - a number is not a struct. But NumPy seems to allow a scalar value for `arr.fill(42)` and `arr[:] = 42` when arr has a compound dtype, somewhat to my surprise.

@tacaswell @aragilar any preference on this?

Reply:
@takluyver I was surprised too. It might just work because `42` converts into `int` and `float` without problem.

Reply:
I think matching numpy makes sense.

Reply:
On the other hand, if this was failing before, I'd expect it to keep failing (unless I missed why it passes now).

Reply:
@aragilar I've limited the dtype forwarding to string types and reverted the changes to the test. This should now work the same as before.

Reply:
@aragilar @takluyver Looks like my latest changes brought back the old behavior. I think we can mark this as resolved.